### ISSUE TITLE
Provider attempts to unmarshal HTTP response errors as JSON

### ISSUE THREAD

**Issue by camelpunch** [mrparkers/terraform-provider-keycloak#239]
Our CI setup deploys a fresh Keycloak and then immediately runs terraform on it. We have an underlying issue where we're not waiting long enough for Keycloak before we run terraform, but debugging that was made a little painful by some confusing terraform provider output:

```
Error: invalid character 'S' looking for beginning of value

  on provider.tf line 4, in provider "keycloak":
   4: provider "keycloak" {
```

Where "S" is actually the first character of the string "Service Unavailable"

[This line](https://github.com/mrparkers/terraform-provider-keycloak/blob/8d54df62b697dd10124dfac9a5d7109c1a0d097d/keycloak/keycloak_client.go#L135) appears to attempt the JSON unmarshalling of that string.

I'm not very familiar with what the appropriate behaviour for a provider is here. Perhaps it would be better to include the HTTP error string in the error message that is returned, replacing the cryptic JSON error message?

Thanks for a very useful provider!

**Comment by mrparkers** [comment]
Yeah, I think we should catch the json unmarshal error and just present the raw string if that fails.  There's no need for the provider to fail here.  Good catch!

**Comment by mrparkers** [comment]
Closed via #304.
